File: templates/starter-python-api/src/path_helper.py
import os
from pathlib import Path

def ensure_dir(env_var: str, default: str) -> Path:
    p = Path(os.getenv(env_var, default)).expanduser()
    p.mkdir(parents=True, exist_ok=True)
    return p

DATA_DIR = ensure_dir("APP_DATA_DIR", "~/.local/share/starter-python-api")
LOG_DIR  = ensure_dir("APP_LOG_DIR",  "~/.cache/starter-python-api/logs")
CACHE_DIR= ensure_dir("APP_CACHE_DIR","~/.cache/starter-python-api")

File: templates/_archive/Root-Arch_251026-0900_GPTT-V1/templates/starter-node-service/src/pathHelper.ts
import { mkdirSync } from 'fs';
import { homedir } from 'os';
import { resolve } from 'path';

function getEnvOrDefault(name: string, fallback: string) {
  return process.env[name] || fallback;
}

export function ensureDirs() {
  const dataDir = resolve(getEnvOrDefault('APP_DATA_DIR', resolve(homedir(), '.local/share/starter-node-service')));
  const logDir  = resolve(getEnvOrDefault('APP_LOG_DIR',  resolve(homedir(), '.cache/starter-node-service/logs')));
  const cacheDir= resolve(getEnvOrDefault('APP_CACHE_DIR',resolve(homedir(), '.cache/starter-node-service')));
  [dataDir, logDir, cacheDir].forEach(d => mkdirSync(d, { recursive: true }));
}
